o.n.core/src/org/netbeans/core/NbAuthenticator.java

```diff
@@ -61,6 +63,7 @@
  */
 final class NbAuthenticator extends java.net.Authenticator {
```

Comment:
Why not just add `@ServiceProvider(service= Authenticator.class)` here?

Reply:
I actually did that initially. (i.e. registering self as a service). But it requires a public no-arg constructor and I didn't have time to analyze why the original author insisted on a private constructor and since it wouldn't actually bring any benefit - besides a bit cleaner code - I then discarded the idea. But I have no objections to do this change.

Reply:
Ah, I see, and `NbAuthenticator` is package private too. I guess this is just standard NetBeans architecture of not over-exposing more than necessary.

I guess:
* we could either make `NbAuthenticator` class and constructor public or
* introduce a `AuthenticatorFactory` which will be registered in the lookup with a default factory producing `NbAuthenticator` or
* use your current solution

Reply:
`java.util.ServiceLoader` requires classes with public constructor so does `Lookup.getDefault()`. Is the class in public or non-public package? If it is in non-public, feel free to make its default constructor public - it won't be considered an API change anyway.